### PR TITLE
fix several MSVC build errors

### PR DESCRIPTION
**Pull request by sherriiiliu**
Fixed a few MSVC (VCToolsVersion=14.0) build errors and warnings
* `DEFINE_string` is a macro and VC compiler complains that it cannot put [ifdef-inside-define](https://stackoverflow.com/questions/5586429/ifdef-inside-define)
* `sleep()` is not a recognizable function. Use `FLAGS_env->SleepForMicroseconds` instead
* Define precise type in comparison to avoid mismatch warning

### CODE REVIEW

File: tools/db_bench_tool.cc [hunk 1]
```diff
@@ -228,6 +225,130 @@ DEFINE_string(
...
+    "\tsstables    -- Print sstable info\n"
+    "\theapprofile -- Dump a heap profile (if supported by this port)\n"
+    "\treplay      -- replay the trace file specified with trace_file\n"
+    "\tgetmergeoperands -- Insert lots of merge records which are a list of "
+    "sorted ints for a key and then compare performance of lookup for another "
+    "key "
+    "by doing a Get followed by binary searching in the large sorted list vs "
+    "doing a GetMergeOperands and binary searching in the operands which are"
+    "sorted sub-lists. The MergeOperator used is sortlist.h\n");
+#endif
```
Comment (siying) [inline tools/db_bench_tool.cc:351]: Can you explain the reason behind this change? This appears to make the code harder to maintain in the future.

Comment (sherriiiliu) [reply tools/db_bench_tool.cc:351]: `DEFINE_string` is a macro and MSVC 14.0 reports a compiler error if we put #ifdef/#ifndef inside #define. The error looks like below:
```
##[error]tools\db_bench_tool.cc(226,0): Error C2121: '#': invalid character: possibly the result of a macro expansion
   223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2121: '#': invalid character: possibly the result of a macro expansion [D:\agent\_work\10\s\build\db_bench.vcxproj]
##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ')' before identifier 'ifndef'
   223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ')' before identifier 'ifndef' [D:\agent\_work\10\s\build\db_bench.vcxproj]
##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ';' before identifier 'ifndef'
   223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ';' before identifier 'ifndef' [D:\agent\_work\10\s\build\db_bench.vcxproj]
##[error]tools\db_bench_tool.cc(226,0): Error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
   223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [D:\agent\_work\10\s\build\db_bench.vcxproj]
##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ';' before identifier 'ROCKSDB_LITE'
   223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ';' before identifier 'ROCKSDB_LITE' [D:\agent\_work\10\s\build\db_bench.vcxproj]
##[error]tools\db_bench_tool.cc(226,0): Error C2059: syntax error: ')'
```
I searched the same issue on stackoverflow [ifdef-inside-define](https://stackoverflow.com/questions/5586429/ifdef-inside-define), [ifdef-inside-a-macro](https://stackoverflow.com/questions/7246512/ifdef-inside-a-macro), and it looks like a general rule but I am not sure why other compilers do not complain. Do you have better idea to make both the code easy to maintain and MSVC compiler happy?

Comment (siying) [reply tools/db_bench_tool.cc:351]: > `DEFINE_string` is a macro and MSVC 14.0 reports a compiler error if we put #ifdef/#ifndef inside #define. The error looks like below:
> 
> ```
> ##[error]tools\db_bench_tool.cc(226,0): Error C2121: '#': invalid character: possibly the result of a macro expansion
>    223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2121: '#': invalid character: possibly the result of a macro expansion [D:\agent\_work\10\s\build\db_bench.vcxproj]
> ##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ')' before identifier 'ifndef'
>    223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ')' before identifier 'ifndef' [D:\agent\_work\10\s\build\db_bench.vcxproj]
> ##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ';' before identifier 'ifndef'
>    223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ';' before identifier 'ifndef' [D:\agent\_work\10\s\build\db_bench.vcxproj]
> ##[error]tools\db_bench_tool.cc(226,0): Error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
>    223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [D:\agent\_work\10\s\build\db_bench.vcxproj]
> ##[error]tools\db_bench_tool.cc(226,0): Error C2146: syntax error: missing ';' before identifier 'ROCKSDB_LITE'
>    223>d:\agent\_work\10\s\tools\db_bench_tool.cc(226): error C2146: syntax error: missing ';' before identifier 'ROCKSDB_LITE' [D:\agent\_work\10\s\build\db_bench.vcxproj]
> ##[error]tools\db_bench_tool.cc(226,0): Error C2059: syntax error: ')'
> ```
> 
> I searched the same issue on stackoverflow [ifdef-inside-define](https://stackoverflow.com/questions/5586429/ifdef-inside-define), [ifdef-inside-a-macro](https://stackoverflow.com/questions/7246512/ifdef-inside-a-macro), and it looks like a general rule but I am not sure why other compilers do not complain. Do you have better idea to make both the code easy to maintain and MSVC compiler happy?

It's OK if there is no alternative.

**Review by siying — APPROVED** [review]

**Comment by siying** [comment]
I'm merging it, but it would be nice if this can be validated in CI. Otherwise, it might be broken by someone else later and we won't be aware of it. I think we have CircleCI's build-windows-vs2017 running (VS 14 is 2017?). Is there a way to add appropriate build options there so that it is covered?

**Comment by sherriiiliu** [comment]
> I'm merging it, but it would be nice if this can be validated in CI. Otherwise, it might be broken by someone else later and we won't be aware of it. I think we have CircleCI's build-windows-vs2017 running (VS 14 is 2017?). Is there a way to add appropriate build options there so that it is covered?

After checking current CircleCI's build-windows-vs2017 CL compile settings(https://circleci.com/api/v1.1/project/github/facebook/rocksdb/162073/output/105/0?file=true&allocation-id=60ed4d53dca05d37afaa2b7d-0-build%2F7A11DAC9), I found the difference is because in CI GFLAGS is disabled while I enabled GFLAGS on my local build. Do you think we can enabled GFLAGS for `db_bench.vcxproj`?

**Pull request closed without merging**